merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -640,8 +641,19 @@ private record ComputedAttributesCodeBlocks(TypeName typeName, FieldSpec fieldDe
             .build());
   }
 
-  private static MethodSpec makeGetDurationTypeMethod(final ActivityTypeRecord activityType) {
+  private MethodSpec makeGetDurationTypeMethod(final ActivityTypeRecord activityType) {
     final var durationParameter = activityType.effectModel().flatMap(EffectModelRecord::durationParameter);
+    if (durationParameter.isPresent()) {
+      for (final var param : activityType.parameters()) {
+        if (param.name.equals(durationParameter.get())) {
+          if (!param.type.toString().equals("gov.nasa.jpl.aerie.merlin.protocol.types.Duration")) {
```

Comment:
Another instance of the hardcoded full type name, if you decide to tackle the other comment. Resolve when the other comment is resolved.

Reply:
Whoops, I don't need to be doing this check twice - I'll go ahead and delete this one.